index.ts

```diff
@@ -108,136 +43,166 @@ class WertWidget {
+
+        break;
+
+      case "position":
+        this.options.listeners?.[event.data.type]?.(event.data.data);
+
+        break;
+
+      case "rate-update":
+        this.options.listeners?.[event.data.type]?.(event.data.data);
 
         break;
 
       case 'close':
         document.body.removeChild(this.iframe);
         document.body.style.overflow = '';
+        this.unListenWidget();
+        this.options.listeners?.[event.data.type]?.();
 
         break;
 
-      default:
+      case "error":
+        this.options.listeners?.[event.data.type]?.(event.data.data);
+
         break;
     }
-
-    const customListener = this.listeners[event.data.type];
-
-    if (customListener) customListener(event.data.data);
   }
 
-  private sendEvent(options: eventOptions): void {
-    if (!options.data) return;
+  private sendEvent(type: string, data: any): void {
+    if (!data) return;
 
-    this.widgetWindow?.postMessage({
-      type: options.type,
-      data: options.data,
-    }, options.origin);
+    this.widgetWindow?.postMessage(
+      { data, type },
+      this.options.origin!
+    );
   }
 
-  getEmbedCode(): string {
+  private getEmbedCode(): string {
     const br = '\n';
     const fileScriptOpen = `<script type="text/javascript" src="${externalStaticOrigin}/wert-${version}.js">`;
-    const scriptEnd = '<' + '/script>'; // eslint-disable-line
+    const scriptEnd = '<' + '/script>';
     const codeScriptOpen = '<script type="text/javascript">';
-    const widgetOptions = {
-      partner_id: this.partner_id,
-      origin: this.origin,
-      ...this.options,
-    };
-    const codeScriptContent1 = `const wertWidget = new WertWidget(${JSON.stringify(widgetOptions, null, 2)});`;
-    const codeScriptContent2 = 'wertWidget.mount();';
-    const code = fileScriptOpen + scriptEnd + br
-      + codeScriptOpen + br
-      + codeScriptContent1 + br
-      + codeScriptContent2 + br
-      + scriptEnd;
-
-    return code;
+    const codeScriptContent1 = `const wertWidget = new WertWidget(${JSON.stringify(
+      this.options,
+      null,
+      2
+    )});`;
+    const codeScriptContent2 = 'wertWidget.open();';
+
+    return fileScriptOpen +
+      scriptEnd +
+      br +
+      codeScriptOpen +
+      br +
+      codeScriptContent1 +
+      br +
+      codeScriptContent2 +
+      br +
+      scriptEnd;
   }
 
   private getEmbedUrl(): string {
     const parametersString = this.getParametersString();
 
-    const url = this.origin + '/' + this.partner_id + '/widget' + parametersString;
-
-    return url;
+    return `${this.options.origin}/${this.options.origin}/widget${parametersString}`;
   }
 
   private getParametersString(): string {
-    const parametersString = Object.entries(this.options)
-      .reduce((accum, [key, value]) => {
-        if (value === undefined) return accum;
+    return Object.entries({
+      ...this.options,
+      widget_layout_mode: this.widget_layout_mode,
```

Comment:
If I'm not mistaken, we don't let our clients pass this param themselves.
If so, why would we need to keep the widgetLayoutMode backward compatible, if we're going to use the newest version of the initializer in stall-spa (i.e. we'll pass the "widget_layout_mode" from now on)?